ddtrace/internal/_encoding.pyx

```diff
@@ -101,6 +103,16 @@ cdef inline int pack_bytes(msgpack_packer *pk, char *bs, Py_ssize_t l):
         ret = msgpack_pack_raw_body(pk, bs, l)
     return ret
 
+cdef inline int pack_bool(msgpack_packer *pk, object n) except? -1:
+    if n is None:
+        return msgpack_pack_nil(pk)
+
+    if PyBool_Check(n):
+        if n:
+            return msgpack_pack_true(pk)
+        else:
+            return msgpack_pack_false(pk)
```

Comment:
You _might_ be able to change this:

```python
cdef inline int pack_bool(msgpack_packer *pk, bint n) except? -1:
    if n:
        return msgpack_pack_true(pk)
    return msgpack_pack_false(pk)
```


It looks like you do an `isinstance(value, bool)` check before calling this function, so we shouldn't have `None`, and then you can have Cython worry about converting from `bool` -> `bint`, and it'll yell and throw a value or type error (I don't remember which) if the values mismatch or if it is unable to convert it.